raiden/src/errors/transaction.rs

```diff
@@ -0,0 +1,162 @@
+use std::fmt;
+
+use thiserror::Error;
+
+const TRANSACTION_CANCELLED_MESSAGE_PREFIX: &str = "Transaction cancelled, please refer cancellation reasons for specific reasons";
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct RaidenTransactionCancellationReasons(pub Vec<Option<RaidenTransactionCancellationReason>>);
+
+impl RaidenTransactionCancellationReasons {
```

Comment:
If you are ok, could you please implement `has_conditional_check_failed` function.
This is because I think it will be used often in app.

Reply:
I implemented these functions.